sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -155,6 +155,11 @@ case class InsertIntoHiveTable(
     val partitionColumns = fileSinkConf.getTableInfo.getProperties.getProperty("partition_columns")
     val partitionColumnNames = Option(partitionColumns).map(_.split("/")).orNull
 
+    // Validate that partition values are specified for partition columns.
+    if (partitionColumnNames != null && partitionColumnNames.size > 0 && partitionSpec.size == 0) {
+      throw new SparkException(ErrorMsg.NEED_PARTITION_ERROR.getMsg)
```

Comment:
@marmbrus Thanks. Actually right after the code block i changed, there are a few places where we raise SparkException. So i thought there may be a reason for it and followed it. :-). I will change all those places as well.